### PR TITLE
Deal with numbers that contain commas

### PR DESCRIPTION
**Pull request by ihough**
Currently if a csv contains "1,605.33` this is imported as 1

### CODE REVIEW

**Comment by seamusabshere** [comment]
@ihough i suggest adding (you need all the funky array stuff in order to memoize false)

```
def number_column?
  return @number_column_query[0] if @number_column_query.is_a?(Array)
  @number_column_query = [model.columns_hash[name.to_s].number?]
  @number_column_query[0]
end
```

then

```
if number_column?
  period_position = value.rindex '.'
  comma_position = value.rindex ','
  if period_position and comma_position and comma_position > period_position
    # uncommon euro style 1.000.000,53
    value = value.delete('.').gsub(',', '.')
  elsif comma_position
    # more common 4,000[.00] style - still don't want the commas
    value = value.delete(',')
  end
end
```

File: test/support/pets.csv [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name,breed,color,age,age_units,weight,height,favorite_food,command_phrase
-Pierre,Tabby,GO,4,years,4.4,30,tomato,"eh"
-Jerry,Beagle,BR/BL,5,years,10,30,cheese,"che"
-Amigo,Spanish Lizarto,GR/BU,17,years,"   ",3,crickets,"    "
-Johnny,Beagle,BR/BL,2,years,20,45,"   ",
+Pierre,Tabby,GO,4,years,4.4,"3.000,5",tomato,"eh"
```
Comment (seamusabshere) [inline test/support/pets.csv:2]: nice - crazy europeans

File: lib/data_miner/attribute.rb [hunk 2]
```diff
@@ -244,6 +250,23 @@ def read(row)
...
       end
       value = value.to_s
+      if number_column?
+        period_position = value.rindex '.'
+        comma_position = value.rindex ','
+        # assume that ',' is a thousands separator and '.' is a decimal point unless we have evidence to the contrary
+        if period_position and comma_position and comma_position > period_position
+          # uncommon euro style 1.000,53
+          value = value.delete('.').gsub(',', '.')
+        elsif comma_position
```
Comment (seamusabshere) [inline lib/data_miner/attribute.rb:263]: can't you collapse this to

```
elsif comma_position and comma_position > (value.length - 4)
  [...]
elsif comma_position
```

Comment (ihough) [inline lib/data_miner/attribute.rb:263]: Will do - you find it easier to read single longer if statement rather than nested ifs?

Comment (seamusabshere) [inline lib/data_miner/attribute.rb:263]: yah - esp. since the final `elsif comma_position` is by far the most common case.

**Pull request merged**